src/Compilers/CSharp/Portable/Symbols/Source/SourceConstructorSymbolBase.cs

```diff
@@ -19,8 +19,9 @@ internal abstract class SourceConstructorSymbolBase : SourceMemberMethodSymbol
         protected SourceConstructorSymbolBase(
             SourceMemberContainerTypeSymbol containingType,
             Location location,
-            CSharpSyntaxNode syntax)
-            : base(containingType, syntax.GetReference(), ImmutableArray.Create(location), SyntaxFacts.HasYieldOperations(syntax))
```

Comment:
>SyntaxFacts.HasYieldOperations(syntax) [](start = 91, length = 38)

Thanks much for fixing this. 

nit: I'm wondering if we could add an assertion to `HasYieldOperations` to either say we're not passing in  type declarations, or we're only passing in method/lambda/local function syntaxes, or that it shouldn't find a descendant node that's a method...

Reply:
Added Assertion